Workspace/src/arm_interface/src/arm_motor_commands_node.cpp

```diff
@@ -0,0 +1,108 @@
+#include <ros/ros.h>
+#include <std_msgs/String.h> //includes string library 
+#include <stdbool.h>
+#include <list>
+#include <vector>
+
+struct msg_from_topics {
+	bool ik_status; 
+	std::list<float> data_points; 
+};
+
+class messageReceivedFromGUIWrapper {
+
+	vector<msg_from_topics> msg;
+
+	void messageReceivedFromGUI(const msg_from_topics values) {
+		msg.add(values);
+	}
+	// To access certain index: msg.at(i); size: msg.size(); remove: msg.pop_back()
+}
+
+	
+
+
+
+// messageRecievedFromGUI return the message that is recieved from the Inverse Kinematics library after it passes the message parameter into the library
+// Not sure if this library exists or not yet
+msg_from_topics messageRecievedFromGUI(const msg_from_topics values) {
+	return values; 
+}
+
+// messageReceieveFromCAN has a message passed into it and returrns the message 
+std::list<float> messageRecievedFromCAN(const std::list<float> values) {
```

Comment:
This is not a ros message. Please see http://wiki.ros.org/std_msgs, you need std_msgs::Float64MultiArray as your message type from CAN